clients/vault/src/issue.rs

```diff
@@ -159,6 +163,74 @@ fn get_issue_id_from_tx_env(tx_env: &TransactionEnvelope) -> Option<IssueId> {
 	}
 }
 
+// Returns oneshot sender for sending status of a ledger.
+//
+// Checks the map if the ledger should be executed/processed.
+// If not, returns None.
+#[doc(hidden)]
+fn create_task_status_sender(
+	processed_map: &mut HashMap<Ledger, SlotTask>,
+	ledger: &Ledger,
+) -> Option<tokio::sync::oneshot::Sender<SlotTaskStatus>> {
+	// An existing task is found.
+	if let Some(existing) = processed_map.get_mut(ledger) {
+		// Only recoverable errors can be given a new task.
+		return existing.recover_with_new_sender()
+	}
+
+	// Not finding the ledger in the map means there's no existing task for this ledger.
+	let (sender, receiver) = tokio::sync::oneshot::channel();
+	tracing::trace!("Creating a task for ledger {}", ledger);
+
+	let ledger_task = SlotTask::new(*ledger, receiver);
+	processed_map.insert(*ledger, ledger_task);
+
+	Some(sender)
```

Comment:
this is now a bit inconsistent with 'ledger' vs 'slot'
you should probably update the comment and variable name too 😅